components/base/wheeled/wheeled_base.go

```diff
@@ -292,6 +300,7 @@ func (base *wheeledBase) WaitForMotorsToStop(ctx context.Context) error {
 }
 
 func (base *wheeledBase) Stop(ctx context.Context, extra map[string]interface{}) error {
+	base.opMgr.CancelRunning(ctx)
```

Comment:
This line looks to be the crux of the PR: most of the rest is benign changes that shouldn't affect the behavior. Have I read this correctly? I'm still new to op managers; why does this line fix the bug?